components/codec/src/columnar/bytes.rs

```diff
@@ -12,37 +12,119 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use bytes_ext::{Buf, BufMut, Bytes};
-use snafu::ResultExt;
+use std::io::{Read, Write};
+
+use bytes_ext::{Buf, BufMut, Bytes, WriterOnBufMut};
+use lz4_flex::frame::{FrameDecoder as Lz4Decoder, FrameEncoder as Lz4Encoder};
+use snafu::{ensure, ResultExt};
 
 use crate::{
     columnar::{
+        Compress, DecodeContext, Decompress, InvalidCompression, InvalidVersion, NotEnoughBytes,
         Result, ValuesDecoder, ValuesDecoderImpl, ValuesEncoder, ValuesEncoderImpl, Varint,
     },
     varint,
 };
 
+/// The layout for the string/bytes:
+/// ```plaintext
+/// +-------------+--------------+------------+-----------------------+-----------------+
+/// | version(u8) | length_block | data_block | length_block_len(u32) | compression(u8) |
+/// +-------------+--------------+------------+-----------------------+-----------------+
+/// ```
+///
+/// Currently, the `compression` has two optional values:
+/// - 0: No compression over the data block
+/// - 1: the data block will be compressed if it is too long
```

Comment:
```suggestion
/// - 1: the data block will be compressed using lz4 if it is too long
```

Reply:
The data block can be compressed by not only lz4.